Doc/library/string.templatelib.rst

```diff
@@ -0,0 +1,313 @@
+   .. describe:: template + other
+                 template += other
+
+       Concatenate this template with another, returning a new
+       :class:`!Template` instance:
+
+       >>> name = "World"
+       >>> list(t"Hello " + t"there {name}!")
+       ['Hello there ', Interpolation('World', 'name', None, ''), '!']
+
+       Concatenation between a :class:`!Template` and a ``str`` is *not* supported.
+       This is because it is ambiguous whether the string should be treated as
+       a static string or an interpolation. If you want to concatenate a
+       :class:`!Template` with a string, you should either wrap the string
+       directly in a :class:`!Template` (to treat it as a static string) or use
+       an :class:`!Interpolation` (to treat it as dynamic):
+
+       >>> from string.templatelib import Template, Interpolation
+       >>> template = t"Hello "
+       >>> # Treat "there " as a static string
+       >>> template += Template("there ")
+       >>> # Treat name as an interpolation
+       >>> name = "World"
+       >>> template += Template(Interpolation(name, "name"))
+       >>> list(template)
+       ['Hello there ', Interpolation('World', 'name', None, '')]
+
+
+.. class:: Interpolation(value, expression="", conversion=None, format_spec="")
+
+   Create a new :class:`!Interpolation` object.
+
+   :param value: The evaluated, in-scope result of the interpolation.
+   :type value: object
+
+   :param expression: The text of a valid Python expression, or an empty string.
+   :type expression: str
+
+   :param conversion: The optional :ref:`conversion <formatstrings>` to be used, one of r, s, and a.
+   :type conversion: ``Literal["a", "r", "s"] | None``
+
+   :param format_spec: An optional, arbitrary string used as the :ref:`format specification <formatspec>` to present the value.
+   :type format_spec: str
+
+   The :class:`!Interpolation` type represents an expression inside a template string.
+
+   :class:`!Interpolation` instances are shallow immutable: their attributes cannot be
+   reassigned.
+
+   .. attribute:: value
+
+       :returns: The evaluated value of the interpolation.
+       :rtype: object
+
+       >>> t"{1 + 2}".interpolations[0].value
+       3
+
+   .. attribute:: expression
+
+       :returns: The text of a valid Python expression, or an empty string.
+       :rtype: str
+
+       The :attr:`~Interpolation.expression` is the original text of the
+       interpolation's Python expression, if the interpolation was created
+       from a t-string literal. Developers creating interpolations manually
+       should either set this to an empty string or choose a suitable valid
+       Python expression.
+
+       >>> t"{1 + 2}".interpolations[0].expression
+       '1 + 2'
+
+   .. attribute:: conversion
+
+       :returns: The conversion to apply to the value, or ``None``.
+       :rtype: ``Literal["a", "r", "s"] | None``
+
+       The :attr:`!Interpolation.conversion` is the optional conversion to apply
```

Comment:
This will be rendered as Interpolation.conversion, so I suggest using double backquotes instead.